bokeh/models/tools.py

```diff
@@ -865,3 +866,89 @@ class RedoTool(Action):
         :height: 18pt
 
     '''
+
+class VertexEditTool(Tool):
+
+    source = Instance(ColumnDataSource, help="""
+    The ColumnDataSource to be modified by the tool.
+    """)
+
+    renderers = List(Instance(Renderer), help="""
+    An explicit list of renderers corresponding to scatter glyphs
+    that may be edited.
+    """)
+
+    x = String('x', help="The name of the x-value column in the supplied source")
+
+    y = String('y', help="The name of the y-value column in the supplied source")
```

Comment:
Add an abstract base class, even if the sole purpose is to group models by functionality. However, given that `x`, `y` are repeated in all new models, you could factor out those two properties to the base class.

Reply:
`source` and `renderers` could be factored out as well.